File: tests/journal.lib/file_sanity.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2023 all rights reserved


// get the journal
#include <pyre/journal.h>
// support
#include <cassert>


// alias the type
using file_t = pyre::journal::file_t;


// exercise the stream device
int main() {
    // the filename
    auto filename = file_t::path_type("file_sanity.out");

    // instantiate
    file_t file(filename);
    // check its name
    assert (file.name() == "file");
    // and its path
    assert (file.path() == filename);

    // all done
    return 0;
}


// end of file

File: tests/journal.pkg/file_sanity.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2023 all rights reserved


def test():
    """
    Verify that we can build a file based device
    """
    # pick a file name
    filename = "file_sanity.log"

    # get the device
    from journal.File import File
    # instantiate
    device = File(path=filename)

    # check its name
    assert device.name == "log"
    # and the path
    assert device.path == filename

    # all done
    return


# main
if __name__ == "__main__":
    # prohibit the journal bindings
    journal_no_libjournal = True
    # run the test
    test()


# end of file
